src/components/Popover/Popover.tsx

```diff
@@ -0,0 +1,204 @@
+import React, {
+  cloneElement,
+  FC,
+  isValidElement,
+  ReactNode,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import { createPortal } from 'react-dom';
+import { usePopper } from 'react-popper';
+import { Placement } from '@popperjs/core';
+import FocusTrap from 'focus-trap-react';
+import classNames from 'classnames';
+import { BrandColor } from '../../types';
+import styles from './Popover.module.scss';
+import Box, { BoxProps } from '../Box/Box';
+
+interface PopoverProps {
+  /**
+   * The Popover is a controlled input, and will be shown when `isOpen === true`.
+   */
+  isOpen: boolean;
+  /**
+   * Color of the arrow background.
+   */
+  arrowColor?: BrandColor;
+  /**
+   * Content of the tooltip. Can be any JSX node, but should be a single node, so it can be assigned a ref.
+   */
+  content?: ReactNode;
+  /**
+   * Padding for the tooltip content. This has a default value
+   * in order to avoid content overlap with the popover arrow.
+   */
+  contentContainerProps?: BoxProps;
+  /**
+   * Callback function to handle when a user clicks outside the Popover
+   */
+  onClickOutside?: () => void;
+  /**
+   * The placement of the Popover relative to its trigger.
+   */
+  placement?: Placement;
+  /**
+   * The target element where the Popover will be portaled to, when `withPortal === true`.
+   */
+  portalTarget?: HTMLElement;
+  /**
+   * Whether you want to trap focus in the Popover element when it is open.
+   * Read more about focus traps:
+   * [Here](https://allyjs.io/tutorials/accessible-dialog.html#trapping-focus-inside-the-dialog)
+   */
+  trapFocus?: boolean;
+  /**
+   * Whether the element should be rendered outside its DOM structure
+   * for reasons of placement. Use this when the element is being cut-off or
+   * re-positioned due to lack of space in the parent container.
+   * NOTE: use `portalTarget` to render the element onto a custom container,
+   * otherwise it will be rendered to the `body` element by default.
+   */
+  withPortal?: boolean;
+  /**
+   * Additional props to be spread to rendered element
+   */
+  [x: string]: any; // eslint-disable-line
+}
+
+const contentContainerDefaults: BoxProps = {
+  background: 'white',
+  padding: 'sm',
+};
+
+const Popover: FC<PopoverProps> = ({
+  arrowColor = undefined,
+  children,
+  content,
+  contentContainerProps = { ...contentContainerDefaults },
+  isOpen,
+  onClickOutside = undefined,
+  placement,
+  portalTarget = document.body,
+  trapFocus = false,
```

Comment:
Should this be true by default?

Reply:
Hey @nathanyoung having the `trapFocus = true` by default is not a problem but it does create a big pain for anybody using our components and running tests. The tests fail to run properly and require an `await` placed strategically, which even after a dev gets it right (ends up delaying tests substantially).

Reply:
ok